### DISCUSSION TITLE
有道翻译笔S6在解锁ADB后的一些发现

### DISCUSSION THREAD

**Discussion by LittleSadSheep** [PenUniverse/PenMods#266] [奇技淫巧]
有道S6，解锁了adb
```
[root@YoudaoDictionaryPen-160:/oem/YoudaoDictPen/output]# cat /etc/issue
Welcome to YD Buildroot
[root@YoudaoDictionaryPen-160:/oem/YoudaoDictPen/output]# uname -a
Linux YoudaoDictionaryPen-160 4.19.164-tag--g98c5e35349c5 #1 SMP PREEMPT Tue Aug 20 17:48:58 CST 2024 armv7l GNU/Linux
[root@YoudaoDictionaryPen-160:/oem/YoudaoDictPen/output]# cat /proc/cpuinfo
processor       : 0
model name      : ARMv7 Processor rev 4 (v7l)
BogoMIPS        : 50.00
Features        : half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm aes pmull sha1 sha2 crc32
CPU implementer : 0x41
CPU architecture: 7
CPU variant     : 0x0
CPU part        : 0xd03
CPU revision    : 4

Hardware        : Cvitek cv182x (Flattened Device Tree)
Revision        : 1826
Serial          : 0000000000000000
[root@YoudaoDictionaryPen-160:/oem/YoudaoDictPen/output]# free -m
              total        used        free      shared  buff/cache   available
Mem:            342         124          31           5         187         203
Swap:           255           0         255
[root@YoudaoDictionaryPen-160:/oem/YoudaoDictPen/output]# df -h
Filesystem                Size      Used Available Use% Mounted on
/dev/root                 1.2G    610.7M    535.3M  53% /
devtmpfs                169.4M         0    169.4M   0% /dev
tmpfs                   171.4M     44.0K    171.4M   0% /dev/shm
tmpfs                   171.4M     64.0K    171.3M   0% /tmp
tmpfs                   171.4M    244.0K    171.2M   0% /run
/dev/mmcblk0p13         124.5M     42.9M     75.2M  36% /userdata
/dev/mmcblk0p14          25.4G      3.2G     20.9G  13% /userdisk
/dev/root                 1.2G    610.7M    535.3M  53% /etc/asound.conf
/dev/mmcblk0p13         124.5M     42.9M     75.2M  36% /etc/input-event-daemon.conf
```
似乎系统的Linux的，SoC是Cvitek cv182x，ArmV7，有342MB的运行内存，但是空闲只有31MB，还有255MB的Swap，感觉折腾的空间不是很大
目前暂时没想出来能折腾些啥，尤其是这小的逆天的运存，空闲空间也小的离谱，不理解网易给翻译笔配了32GB储存空间是干嘛的，感觉就算能装进去点啥也会因为运存太小而用不了

**Comment by LittleSadSheep** [comment]
翻了一遍没翻出来有软件包管理器，无论是apt rpm yum还是pm都用不了

**Comment by ovo-Tim** [comment]
我补充一下，S6 pro(题主是 S6) 的根目录已经是只读状态了，这意味着只有 chroot 可以折腾折腾。

X server 貌似可以跑起来(chroot)，这样的话……

**Reply by SkySight-666** [reply]
> 我补充一下，S6 pro(题主是 S6) 的根目录已经是只读状态了，这意味着只有 chroot 可以折腾折腾。
> 
> X server 貌似可以跑起来(chroot)，这样的话……

remount即可

**Reply by ovo-Tim** [reply]
@SkySight-666 你确定吗？文件系统貌似是 erofs,我刚测试也不行
![image](https://github.com/user-attachments/assets/dee5c5e3-7e08-4f70-95c3-2ea63556d58e)

**Reply by SkySight-666** [reply]
> @SkySight-666 你确定吗？文件系统貌似是 erofs,我刚测试也不行 ![image](https://private-user-images.githubusercontent.com/116942368/423196528-dee5c5e3-7e08-4f70-95c3-2ea63556d58e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDIxMTMxNDEsIm5iZiI6MTc0MjExMjg0MSwicGF0aCI6Ii8xMTY5NDIzNjgvNDIzMTk2NTI4LWRlZTVjNWUzLTdlMDgtNGY3MC05NWMzLTJlYTYzNTU2ZDU4ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMzE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDMxNlQwODE0MDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01OTI3ZWI1OGMzZTU0NTI1Njg1YmExOTFmMjJjOWMyMTRhMGZmNWUwOWVmOWE4ZGIzYTUxMTNkZDgxOTE2NjEyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.L1r1u9oKW4I9DM4y6wTeBmT7UM6yAgEC1VkwEktWdYU)

稍等

**Reply by SkySight-666** [reply]
> > 我补充一下，S6 pro(题主是 S6) 的根目录已经是只读状态了，这意味着只有 chroot 可以折腾折腾。
> > X server 貌似可以跑起来(chroot)，这样的话……
> 
> remount即可

哦哦你是S6p

**Reply by SkySight-666** [reply]
> > @SkySight-666 你确定吗？文件系统貌似是 erofs,我刚测试也不行 ![image](https://private-user-images.githubusercontent.com/116942368/423196528-dee5c5e3-7e08-4f70-95c3-2ea63556d58e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDIxMTMxNDEsIm5iZiI6MTc0MjExMjg0MSwicGF0aCI6Ii8xMTY5NDIzNjgvNDIzMTk2NTI4LWRlZTVjNWUzLTdlMDgtNGY3MC05NWMzLTJlYTYzNTU2ZDU4ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMzE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDMxNlQwODE0MDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01OTI3ZWI1OGMzZTU0NTI1Njg1YmExOTFmMjJjOWMyMTRhMGZmNWUwOWVmOWE4ZGIzYTUxMTNkZDgxOTE2NjEyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.L1r1u9oKW4I9DM4y6wTeBmT7UM6yAgEC1VkwEktWdYU)
> 
> 稍等

那应该就是不行 
有道搞什么

**Reply by ovo-Tim** [reply]
顺便预告一下，我快跑起来浏览器了（
(其实 chromium 本来就跑得起来，但是 xorg 识别的屏幕是竖着的，所以完全没法用)

**Reply by SkySight-666** [reply]
> 顺便预告一下，我快跑起来浏览器了（ (其实 chromium 本来就跑得起来，但是 xorg 识别的屏幕是竖着的，所以完全没法用)

？
这玩意是标准屏幕输出？

**Reply by ovo-Tim** [reply]
> > 顺便预告一下，我快跑起来浏览器了（ (其实 chromium 本来就跑得起来，但是 xorg 识别的屏幕是竖着的，所以完全没法用)
> 
> ？ 这玩意是标准屏幕输出？

framebuffter， 只能跑 Xorg 跑不起来 wayland，而且旋转不了屏幕，我只能写一个旋转过的浏览器（

**Reply by SkySight-666** [reply]
> 顺便预告一下，我快跑起来浏览器了（ (其实 chromium 本来就跑得起来，但是 xorg 识别的屏幕是竖着的，所以完全没法用)

S6p内存夺少
S6内存300多m

**Reply by ovo-Tim** [reply]
> > 顺便预告一下，我快跑起来浏览器了（ (其实 chromium 本来就跑得起来，但是 xorg 识别的屏幕是竖着的，所以完全没法用)
> 
> S6p内存夺少 S6内存300多m

我也是，只能开 swap 了（

**Reply by SkySight-666** [reply]
> > > 顺便预告一下，我快跑起来浏览器了（ (其实 chromium 本来就跑得起来，但是 xorg 识别的屏幕是竖着的，所以完全没法用)
> > 
> > 
> > ？ 这玩意是标准屏幕输出？
> 
> framebuffter， 只能跑 Xorg 跑不起来 wayland，而且旋转不了屏幕，我只能写一个旋转过的浏览器（

好强。
想问几个问题：
1. Xorg以及chromium应该是运行在chroot上的？
2. 如何解决有道自己的输出冲突
3. 有触摸吗
已经快触及我的知识盲区了 大佬好强

**Reply by ovo-Tim** [reply]
> > > > 顺便预告一下，我快跑起来浏览器了（ (其实 chromium 本来就跑得起来，但是 xorg 识别的屏幕是竖着的，所以完全没法用)
> > > 
> > > 
> > > ？ 这玩意是标准屏幕输出？
> > 
> > 
> > framebuffter， 只能跑 Xorg 跑不起来 wayland，而且旋转不了屏幕，我只能写一个旋转过的浏览器（
> 
> 好强。 想问几个问题：
> 
> 1. Xorg以及chromium应该是运行在chroot上的？
> 2. 如何解决有道自己的输出冲突
> 3. 有触摸吗
>    已经快触及我的知识盲区了 大佬好强

哦对，我该把这些分享出来的，等我一会开个仓库
1. 是
2. 我找到了关掉自带服务和保留无线的办法
3. 昨天一个下午的工作（

**Reply by SkySight-666** [reply]
> > > > > 顺便预告一下，我快跑起来浏览器了（ (其实 chromium 本来就跑得起来，但是 xorg 识别的屏幕是竖着的，所以完全没法用)
> > > > 
> > > > 
> > > > ？ 这玩意是标准屏幕输出？
> > > 
> > > 
> > > framebuffter， 只能跑 Xorg 跑不起来 wayland，而且旋转不了屏幕，我只能写一个旋转过的浏览器（
> > 
> > 
> > 好强。 想问几个问题：
> > 
> > 1. Xorg以及chromium应该是运行在chroot上的？
> > 2. 如何解决有道自己的输出冲突
> > 3. 有触摸吗
> >    已经快触及我的知识盲区了 大佬好强
> 
> 哦对，我该把这些分享出来的，等我一会开个仓库
> 
> 1. 是
> 2. 我找到了关掉自带服务和保留无线的办法
> 3. 昨天一个下午的工作（

强佬，谢谢

**Reply by ovo-Tim** [reply]
> > > > > > 顺便预告一下，我快跑起来浏览器了（ (其实 chromium 本来就跑得起来，但是 xorg 识别的屏幕是竖着的，所以完全没法用)
> > > > > 
> > > > > 
> > > > > ？ 这玩意是标准屏幕输出？
> > > > 
> > > > 
> > > > framebuffter， 只能跑 Xorg 跑不起来 wayland，而且旋转不了屏幕，我只能写一个旋转过的浏览器（
> > > 
> > > 
> > > 好强。 想问几个问题：
> > > 
> > > 1. Xorg以及chromium应该是运行在chroot上的？
> > > 2. 如何解决有道自己的输出冲突
> > > 3. 有触摸吗
> > >    已经快触及我的知识盲区了 大佬好强
> > 
> > 
> > 哦对，我该把这些分享出来的，等我一会开个仓库
> > 
> > 1. 是
> > 2. 我找到了关掉自带服务和保留无线的办法
> > 3. 昨天一个下午的工作（
> 
> 强佬，谢谢

基于前人的工作 :hugs:

**Reply by wyxdlz54188** [reply]
哥有空来我们Q群开发miniapp吗
需要C/C++ 和 vuejs

**Reply by wyxdlz54188** [reply]
如果可以的话请联系liyuqili_1379zi@qq.com

**Reply by ovo-Tim** [reply]
@wyxdlz54188 最近折腾动力不大(因为学校可以带电脑了)，不过开发软件难度比较大吧，你们有思路吗？我的建议是直接把原来的服务干掉，自己跑浏览器什么的，这样比较方便，可玩性也高

**Reply by wyxdlz54188** [reply]
我们可以多开app，安装自己的应用。难度也是有点的，至少目前为止不停的在修环境上的错误

**Reply by wyxdlz54188** [reply]
详情邮箱私信加群QQ具体进度暂时不方便公开

**Comment by ovo-Tim** [comment]
可以看看[我跑图形化界面的教程](https://github.com/ovo-Tim/Youdao-DictPenS6P)
